### ISSUE TITLE
New lint on capitalized acronym

### ISSUE THREAD

**Issue by matsujika** [rust-lang/rust-clippy#6300]
### What it does

Warns on uses of a capitalized acronym(e.g. `HTTP` not `Http`) in camel case idents. 

### Categories (optional)

- Kind: `style`(?)

Though capitalized acronyms are against [naming convention](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html) and might be mistakenly practiced by users, nor rustc and clippy doesn't have a lint for this.

### Drawbacks

None.

### Example

```rust
enum HTTPProtocol {}
```

Could be written as:

```rust
enum HttpProtocol {}
```

I'm not yet sure this lint should be in rustc or in clippy so I need some advice. I've found [nonstander_style.rs from rustc_lint](https://github.com/rust-lang/rust/blob/master/compiler/rustc_lint/src/nonstandard_style.rs) has similar lints, FYI.

**Comment by giraffate** [comment]
related issue: https://github.com/rust-lang/rust-clippy/issues/1335

**Comment by ebroto** [comment]
Thanks for taking the time to file the issue.

I'm closing this as duplicate of #1335 because that one has a bit more info about false positives.
